cartridges/int_algolia/cartridge/scripts/algolia/helper/modelHelper.js

```diff
@@ -47,7 +51,15 @@ function getColorVariations(product, locale) {
                     colorValue.value
                 ).toString(),
                 color: colorValue.displayValue,
-            });
+            };
+
+            if (IS_PWA) {
+                var variantID = getVariantID(variationModel, colorVariationAttribute, colorValue);
+
+                variationObject.variantID = variantID; // Required to create product detail page URL in PWA
```

Comment:
Here you will have a variant with a random size selected (the first that matches the color).
On the official demo, I see that the URLs of the variants are built using the color code (`colorValue.value`):
https://pwa-kit.mobify-storefront.com/global/en-GB/product/25565284M?color=JJ169XX
https://pwa-kit.mobify-storefront.com/global/en-GB/product/25565284M?color=JJ3KRXX

This permits to have the color selected but not the size. Can't we add a `colorCode` attribute and use it to do the same?

Reply:
I see your point, let me check it.